### PR TITLE
Prepend database-level logging with database name

### LINKED ISSUE #3815: Include db and handler context in all db-specific logging
Diagnosing issues for Sync Gateway nodes running multiple databases is more difficult for log messages that don't otherwise identify a database.  Including the database name in log messages in a standard way would address this.

- [ ] Pass the blip context id or REST handler serial number down to changes processing, for inclusion on changes-related logging
- [ ] Include database identifier for changes and crud logging

### PR DESCRIPTION
**Pull request by adamcfraser**
Fixes #3815 

Sample modified log format (where 'default' is the db name):
```
2018-11-05T15:21:04.304-08:00 [INF] HTTP:  #001: GET /default/_changes (as ADMIN)
2018-11-05T15:21:04.304-08:00 [INF] Changes: [default] MultiChangesFeed(channels: {*}, options: {Since:0 Limit:0 Conflicts:false IncludeDocs:false Wait:false Continuous:false Terminator:0xc0001eaba0 HeartbeatMs:0 TimeoutMs:300000 ActiveOnly:false}) ... 
2018-11-05T15:21:04.344-08:00 [INF] Changes: [default] MultiChangesFeed done 
```

### CODE REVIEW

**Comment by adamcfraser** [comment]
@bbrks Would like your review on this when you're back - not urgent.

File: db/repair_bucket.go [hunk 1]
```diff
@@ -123,7 +123,7 @@ This is how the view is iterated:
                                                     NumProcessed: 0
 
 * It starts with an empty start key
-* For the next page, it uses the last key processed as the new start key
+* For the next pageo, it uses the last key processed as the new start key
```
Comment (JFlath) [inline db/repair_bucket.go]: Unrelated change/typo?

Comment (JFlath) [reply db/repair_bucket.go]: Ooh, let me try this thing...

```suggestion
* For the next page, it uses the last key processed as the new start key
```

**Review by bbrks — CHANGES_REQUESTED** [review]
Missing a necessary return in the if `context == nil` blocks to prevent panics/double logging.

See what you think about the other comments around exiting early too.

**Comment by bbrks** [comment]
Closing this PR in favour of a more generic approach to contextual logging. See last update on #3815

**Pull request closed without merging**